doc/ROADMAP.rst.txt

```diff
@@ -0,0 +1,377 @@
+added.  Other things that are needed:
+
+  - deprecate the ``fmin_*`` functions in the documentation, ``minimize`` is
+    preferred.
+  - clearly define what's out of scope for this module.
+
+
+signal
+``````
+*Convolution and correlation*: (Relevant functions are convolve, correlate,
+fftconvolve, convolve2d, correlate2d, and sepfir2d.) Eliminate the overlap with
+`ndimage` (and elsewhere).  From `numpy`, `scipy.signal` and `scipy.ndimage`
+(and anywhere else we find them), pick the "best of class" for 1-D, 2-D and n-d
+convolution and correlation, put the implementation somewhere, and use that
+consistently throughout scipy.
+
+*B-splines*: (Relevant functions are bspline, cubic, quadratic, gauss_spline,
+cspline1d, qspline1d, cspline2d, qspline2d, cspline1d_eval, and spline_filter.)
+Move the good stuff to `interpolate` (with appropriate API changes to match how
+things are done in `interpolate`), and eliminate any duplication.
+
+*Filter design*: merge `firwin` and `firwin2` so `firwin2` can be removed.
+
+*Continuous-Time Linear Systems*: remove `lsim2`, `impulse2`, `step2`.  Make
+`lsim`, `impulse` and `step` "just work" for any input system.  Improve
+performance of ltisys (less internal transformations between different
+representations).
+
+*Wavelets*: what's there now doesn't make much sense.  Continous wavelets
+only at the moment - decide whether to completely rewrite or remove them.
+Discrete wavelet transforms should be added, see the "new modules" section
+below.
+
+
+sparse
+``````
+The sparse matrix formats are getting feature-complete but are slow ...
+reimplement parts in Cython?
+
+    - Small matrices are slower than PySparse, needs fixing
+
+There are a lot of formats.  These should be kept, but
+improvements/optimizations should go into CSR/CSC, which are the preferred
+formats.  LIL may be the exception, it's inherently inefficient.  It could be
+dropped if DOK is extended to support all the operations LIL currently
+provides.
+
+Don't emulate np.matrix behavior, drop 2-D?
+
+
+sparse.csgraph
+``````````````
+This module is in good shape.
+
+
+sparse.linalg
+`````````````
+Arpack is in good shape.
+
+isolve:
+
+    - callback keyword is inconsistent 
+    - tol keyword is broken, should be relative tol
+    - Fortran code not re-entrant (but we don't solve, maybe re-use from
+      PyKrilov)
+
+dsolve:
+
+    - add sparse Cholesky or incomplete Cholesky
+    - look at CHOLMOD
+
+
+Ideas for new features:
+
+    - Wrappers for PROPACK for faster sparse SVD computation.
+
+spatial
+```````
+KDTree/cKDTree and the QHull wrappers are in good shape.  The distance module
+needs bug fixes in the distance metrics, and distance_wrap.c needs to be
+cleaned up (maybe rewrite in Cython).
+
+
+special
+```````
+special has a lot of functions that need improvements in precision.  All
+functions that are also implemented in mpmath can be tested against mpmath, and
```

Comment:
Though there are still a lot of functions that need improvements in precision, probably the only show-stoppers are hypergeometric functions, parabolic cylinder functions, and spheroidal wave functions. Three possible ways to handle this:

 - Get good double-precision implementations. This is doable for parabolic cylinder functions (in progress). I think it's possible for hypergeometric functions, though maybe not in time. For spheroidal wavefunctions this is not possible with current theory.

- Port Boost's arbitrary precision library and use it under the hood to get double precision accuracy. This might be necessary as a stopgap measure for hypergeometric functions; the idea of using arbitrary precision has been suggested before by @nmayorov and in https://github.com/scipy/scipy/issues/5349#issuecomment-244542364. I think this is necessary for spheroidal wave functions, but thankfully someone has already done it: https://github.com/radelman/scattering.

- Add clear warnings to the documentation about the limits of the existing implementations.